syncstorage/tests/functional/test_storage.py

```diff
@@ -73,27 +73,37 @@ def _switch_user(self):
         finally:
             self.root = orig_root
 
+    def retry(self, method, *args, **kwargs):
```

Comment:
Another option is monkey patching self.app.do_request (which is already done to setup auth tokens here: https://github.com/mozilla-services/server-syncstorage/blob/master/syncstorage/tests/functional/support.py#L42). Could only apply if `req.method not in ('GET', 'HEAD')`. It would reduce this patch to only a few lines

Reply:
I think the loop should strictly match what [the decorator does](https://github.com/mozilla-services/server-syncstorage/blob/master/syncstorage/views/decorators.py#L62): just 1 retry.

Reply:
I'm not a fan of monkey patching, mostly because it's too easy to lose the patch unless you know to look for it. 

I'll fix up `retry` to use the same time interval and single retry like you pointed out.

Reply:
I'm personally not a huge fan of string-named method calls like this, but it's purely a stylistic thing so I won't push back on it.  You could also consider making wrappers like `self.delete_with_retry`, self.post_json_with_retry` or similar that might read a little more cleanly.

Reply:
I'll be honest, the `retry(func,..` was mostly just to make the find/replace a lot easier, and to limit the patch to just one bottleneck function. I can see the value of clearly declared function names, though.